nuclia/sdk/kb.py

```diff
@@ -552,6 +612,65 @@ async def del_label(self, *, labelset: str, label: str, **kwargs):
             kbid=ndb.kbid, labelset=labelset, content=labelset_obj
         )
 
+    @kb
+    async def get_graph(
+        self,
+        uid: Optional[str] = None,
+        slug: Optional[str] = None,
+        **kwargs,
+    ):
+        res = await self.resource.get(
+            ndb=kwargs["ndb"], rid=uid, slug=slug, show=["basic", "relations"]
+        )
+        return res.usermetadata.relations
+
+    @kb
+    async def add_graph(
+        self,
+        graph: List[Union[GraphRelation, dict]],
+        slug: Optional[str] = None,
+        **kwargs,
+    ):
+        kw = {
+            "ndb": kwargs["ndb"],
+            "usermetadata": {
+                "relations": [
+                    get_relation(relation).to_relation() for relation in graph
+                ]
+            },
+        }
+        if slug:
+            kw["slug"] = slug
+        await self.resource.create(**kw)
+
+    @kb
+    async def update_graph(
+        self,
+        graph: List[Union[GraphRelation, dict]],
+        uid: Optional[str] = None,
+        slug: Optional[str] = None,
+        override: Optional[bool] = False,
+        **kwargs,
+    ):
+        new_relations = [get_relation(relation).to_relation() for relation in graph]
+        if not override:
+            relations = await self.get_graph(uid=uid, slug=slug, **kwargs)
+            relations.extend(new_relations)
```

Comment:
Are we ok if there are duplicate relations? Is it on us to verify, or on the client side?

Reply:
I would say users will check themselves.